nautilus: qa/tasks/{ceph,ceph_manager}: drop py2 support

ideepika opened the pull request:
cherry pick for: a7f18e4 from  https://github.com/ceph/ceph/pull/35817 as #37690  depends on it.
fixes failures seen in: https://pulpito.ceph.com/?branch=wip-yuri8-testing-2020-10-26-0758-octopus, needed for #37690 backport

```
File "/home/teuthworker/src/github.com_ceph_ceph-c_wip-yuri8-testing-2020-10-26-0758-octopus/qa/tasks/ceph.py", line 161, in invoke_logrotate
    stderr=StringIO()
NameError: name 'StringIO' is not defined
2020-10-26T22:58:41.460 INFO:teuthology.orchestra.run.smithi063:> adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage ceph --cluster ceph osd dump --format=json
```

see also: https://github.com/ceph/ceph/pull/37863


<!--
Thank you for opening a pull request!  Here are some tips on creating
a well formatted contribution.

Please give your pull request a title like "[component]: [short description]"

This is the format for commit messages:

"""
[component]: [short description]

[A longer multiline description]

Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
Signed-off-by: [Your Name] <[your email]>
"""

The Signed-off-by line is important, and it is your certification that
your contributions satisfy the Developers Certificate or Origin.  For
more detail, see SubmittingPatches.rst.

The component is the short name of a major daemon or subsystem,
something like "mon", "osd", "mds", "rbd, "rgw", etc. For ceph-mgr modules,
give the component as "mgr/<module name>" rather than a path into pybind.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview.  More information for contributors is available here:
https://raw.githubusercontent.com/ceph/ceph/master/SubmittingPatches.rst

-->
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

ajarr reviewed (changes requested):
Failures introduced,
https://pulpito.ceph.com/yuriw-2020-10-30_19:26:24-kcephfs-wip-yuri7-testing-2020-10-30-1058-nautilus-distro-basic-smithi/5576271/
https://pulpito.ceph.com/yuriw-2020-10-30_19:26:24-kcephfs-wip-yuri7-testing-2020-10-30-1058-nautilus-distro-basic-smithi/5576303/

tchaikov commented:
@ideepika could you backport #33805 to nautilus in this changeset as well? i think it addresses the failures mentioned in https://github.com/ceph/ceph/pull/37906#pullrequestreview-521532541 (the above comment).

ideepika commented:
> @ideepika could you backport #33805 to nautilus in this changeset as well? i think it addresses the failures mentioned in https://github.com/ceph/ceph/pull/37906#pullrequestreview-521532541 (the above comment).

Yes, sure! Thanks for taking a look!

ideepika commented:
@tchaikov updated, thanks!

ideepika commented on qa/tasks/ceph_manager.py:
```diff
@@ -23,7 +24,8 @@
 from teuthology.orchestra import run
 from teuthology.exceptions import CommandFailedError
-import six
+from tasks.thrasher import Thrasher
```
@tchaikov I see some missing code that needs to be backported because of which I am seeing https://jenkins.ceph.com/job/ceph-api/6116/ failure should I backport 8623bd33d0f85ba529d2808cc44cc1dafd32d1b1 to this PR itself?

tchaikov commented on qa/tasks/ceph_manager.py:
```diff
@@ -241,8 +241,8 @@ def kill_osd(self, osd=None, mark_down=False, mark_out=False):
                     proc = exp_remote.run(args=['type', 'ceph-objectstore-tool'], 
-                               wait=True, check_status=False, stdout=BytesIO(),
-                               stderr=BytesIO())
+                               wait=True, check_status=False, stdout=StringIO(),
```
please explain why this change is not cherry-picked from master in the commit message.

ideepika replied:
sorry, tried to address failure earlier; and then force pushed the updated commits after syncing with you. Seems like the previous cherry picks were not removed.

ideepika commented:
rados
http://pulpito.front.sepia.ceph.com/ideepika-2020-12-11_19:23:47-rados-wip-deepika-testing-2020-12-08-2006-nautilus-distro-basic-smithi/
rerun
http://pulpito.front.sepia.ceph.com/ideepika-2020-12-15_10:04:42-rados-wip-deepika-testing-2020-12-08-2006-nautilus-distro-basic-smithi/

yuriw commented:
![](https://github.trello.services/images/mini-trello-icon.png) [wip-yuri-testing-2021-01-18-0834-nautilus](https://trello.com/c/SacHTFSm/1113-wip-yuri-testing-2021-01-18-0834-nautilus)

yuriw commented:
@ideepika this passed tests in our batch, so pls merge at will

ideepika commented:
@ajarr can you review, if is relevant

tchaikov commented on qa/tasks/ceph_manager.py:
```diff
@@ -260,7 +260,7 @@ def kill_osd(self, osd=None, mark_down=False, mark_out=False):
                     proc = exp_remote.run(args=cmd, wait=True,
                                           check_status=False,
-                                          stdout=BytesIO(), stderr=BytesIO())
+                                          stdout=StringIO(), stderr=StringIO())
```
could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes

ideepika replied:
sure, updated!

smithfarm commented:
jenkins test make check

smithfarm commented:
@yuriw This needs another round of testing because it was force-pushed since the last run.

tchaikov commented:
please cherry pick using `git cherry-pick -x`. and the title of a commit message should summarize the change made in at commit. if a commit is not cherry-picked from master, you might want to explain the rationale in the commit message.

ideepika commented:
> @yuriw This needs another round of testing because it was force-pushed since the last run.

@smithfarm just addressed : https://github.com/ceph/ceph/pull/37906#discussion_r562258143

smithfarm reviewed (changes requested):
@ideepika Sorry to be such a pain, but when I first reviewed this I apparently overlooked that this PR contains cherry picks that are missing the `(cherry picked from commit ...)` line which is added by `git cherry-pick -x`.

So, please fix the commit messages of those commits that are cherry-picks. The commit message must show the SHA1 of the master commit that was cherry-picked.

For more information on this topic, please read https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst and especially https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst#cherry-picking-rules ... this document attempts to explain not only the "how", but also the "why" :-)

ideepika commented:
> @ideepika Sorry to be such a pain, but when I first reviewed this I apparently overlooked that this PR contains cherry picks that are missing the `(cherry picked from commit ...)` line which is added by `git cherry-pick -x`.
> 
> So, please fix the commit messages of those commits that are cherry-picks. The commit message must show the SHA1 of the master commit that was cherry-picked.
> 
> For more information on this topic, please read https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst and especially https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst#cherry-picking-rules ... this document attempts to explain not only the "how", but also the "why" :-)

@smithfarm sorry, missed notification. No worries at all I can amend to commits and thanks for observing that, I might have missed somehow

tchaikov commented:
@ideepika "qa/tasks/ceph_manager: wrt master, we have moved to using run_ceph_objectstore_tool which uses"

this is not the ideal title for a commit message. please refer to https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#commit-title

ideepika commented:
> @ideepika "qa/tasks/ceph_manager: wrt master, we have moved to using run_ceph_objectstore_tool which uses"
> 
> this is not the ideal title for a commit message. please refer to https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#commit-title

@tchaikov that makes sense, I updated `qa/tasks/ceph_manager: use s/ByteIO/StringIO in stdout for ceph-objectstore-tool` would appreciate what you think! thanks!

yuriw commented:
test this please
